amxmodx/messages.h

```diff
@@ -50,52 +48,45 @@ class RegisteredMessage
 		{
 			this->m_Cleanup = true;
 
-			CVector<int>::iterator iter = m_Forwards.begin();
-			CVector<int>::iterator end = m_Forwards.end();
-			while (iter != end)
+			for (size_t i = 0; i < m_Forwards.length(); ++i)
 			{
-				if (*iter == fwd)
+				int& forward = m_Forwards[i];
+
+				if (forward == fwd)
 				{
-					if (*iter != -1)
+					if (forward != -1)
 					{
-						unregisterSPForward(*iter);
+						unregisterSPForward(forward);
 					}
-					*iter = -1;
+
+					forward = -1;
+
 					return true;
 				}
-				else
-				{
-					iter++;
-				}
 			}
-		
 		}
 		else
 		{
-			CVector<int>::iterator iter = m_Forwards.begin();
-			CVector<int>::iterator end = m_Forwards.end();
-			while (iter != end)
+			for (size_t i = 0; i < m_Forwards.length(); ++i)
 			{
-				if (*iter == fwd)
+				int forward = m_Forwards[i];
+
+				if (forward == fwd)
 				{
-					if (fwd != -1)
+					if (forward != -1)
 					{
-						unregisterSPForward(fwd);
+						unregisterSPForward(forward);
 
-						m_Forwards.erase(iter);
+						m_Forwards.remove(forward);
 
 						return true;
 					}
 					else
 					{
 						// -1 could be in here more than once
-						m_Forwards.erase(iter);
```

Comment:
Worth to note (for changelog) that there was a bug here. It removes an iteration without setting `iter` to the new value and with the wrong loop condition where vector should be checked if empty instead.

If you pass there, this results in an infinite loop. It can happen when you call `unregister_message` native in its own hook.